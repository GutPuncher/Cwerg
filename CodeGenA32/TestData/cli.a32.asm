# cli

.fun main NORMAL [U32] = [U32 A32]

.bbl start
    poparg argc:U32
    poparg argv:A32

    pusharg argc
    bsr print_u_ln

    mov i:U32 0
    bra check
.bbl loop
    shl index:U32 i 2
    ld  arg:A32 argv index
    pusharg arg
    bsr print_s_ln
    add i i 1
.bbl check
    blt i argc loop
    pusharg 0:U32
    ret
