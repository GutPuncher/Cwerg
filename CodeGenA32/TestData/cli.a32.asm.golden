7
build/cli.asm.a32.exe
1
2
3
aa
bbb
ccc
